### PR TITLE
[WIP] Fix LightGBM and add test

### LINKED ISSUE #1424: Incorrect model returned by LightGBM calibration for multiclass with softmax
https://github.com/dotnet/machinelearning/blob/5123aeedb91a435236e8b94ba0f17a6d15a20db5/src/Microsoft.ML.LightGBM/LightGbmMulticlassTrainer.cs#L104

The model returned here is only correct if the objective type is `multiclassova`, since the `OvaPredictor` predictor type created here always just normalises the `_numClass` probabilities to sum to one:
https://github.com/dotnet/machinelearning/blob/5123aeedb91a435236e8b94ba0f17a6d15a20db5/src/Microsoft.ML.StandardLearners/Standard/MultiClass/Ova.cs#L575

However, when the objective type is `multiclass` (i.e., softmax), what should happen is that the `OvaPredictor` type should be given an array of predictors that return the _raw_ scores from each of the trees (i.e., the result of `CreateBinaryPredictor` instead of the `FeatureWeightsCalibratedPredictor`), and instead of normalising the outputs of these predictors, it should be computing the softmax function of the raw scores, as is done in the LightGBM implementation:
https://github.com/Microsoft/LightGBM/blob/087d30623aef4b50592cf528eb50c1c55baeaab8/src/objective/multiclass_objective.hpp#L115

These kind of bugs would be easily detected if there were unit tests comparing the results of evaluating the managed ML.NET GBDTs against the original native implementation (i.e., calling `LGBM_BoosterPredictForMat` on the Booster object used to calibrate the model).

### PR DESCRIPTION
**Pull request by wschin**
Fixes #1424. The first commit only creates some doc strings created when I felt very struggling when trying to reuse LightGBM facilities in ML.NET.

### CODE REVIEW

File: src/Microsoft.ML.Core/Utilities/MathUtils.cs [hunk 1]
```diff
@@ -507,7 +507,12 @@ public static Float Tanh(Float x)
         /// </summary>
         public static Float SigmoidSlow(Float x)
         {
-            return 1 / (1 + ExpSlow(-x));
+            // The following two return statements are equivalent but one uses e^(-x) and the other one uses e^x.
```
Comment (Zruty0) [inline src/Microsoft.ML.Core/Utilities/MathUtils.cs]: >The following two return statements are equivalent [](start = 15, length = 50)

I would rephrase it with 
```
// The following two expressions are mathematically equivalent.
// Due to the arithmetic precision issues we should not call exp(x) for large positive x: instead, we call exp(-x).
```

File: src/Microsoft.ML.LightGBM/WrappedLightGbmDataset.cs [hunk 2]
```diff
@@ -17,6 +17,27 @@ internal sealed class Dataset : IDisposable
         private int _lastPushedRowID;
         public IntPtr Handle => _handle;
 
+        /// <summary>
+        /// Create a <see cref="Dataset"/> for storing training and prediciton data under LightGBM framework. The main goal of this function
+        /// is not marshling ML.NET data set into LightGBM format but just creates a (unmanaged) container where examples can be pushed into by calling
```
Comment (Zruty0) [inline src/Microsoft.ML.LightGBM/WrappedLightGbmDataset.cs]: >marshling [](start = 19, length = 9)

marshaling

File: test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs [hunk 3]
```diff
@@ -233,5 +237,150 @@ public void LightGbmMultiClassEstimator()
...
+            public uint Label;
+            [VectorType(_classNumber)]
+            public float[] Score;
+        }
+
+        [BestFriend]
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))] // LightGBM is 64-bit only
+        public void LightGbmMultiClassEstimatorCompareOva()
+        {
+            //////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (Zruty0) [inline test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs]: > ////////////////////////////////////////////////////////////////////////////////////////////// [](start = 11, length = 95)

we don't use such separator comments in ML.NET.

File: test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs [hunk 4]
```diff
@@ -233,5 +237,150 @@ public void LightGbmMultiClassEstimator()
...
+                    sum += MathUtils.SigmoidSlow(sigmoidScale* (float)nativeResult1[j + i * _classNumber]);
+                }
+                for (int j = 0; j < _classNumber; ++j)
+                {
+                    double prob = MathUtils.SigmoidSlow(sigmoidScale * (float)nativeResult1[j + i * _classNumber]);
+                    Assert.Equal(prob / sum, predictions[i].Score[j], 6);
+                }
+            }
+
+        }
```
Comment (Zruty0) [inline test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs]: >} [](start = 8, length = 1)

Even though we are not making baseline comparisons, please call `Done()` at the end of every test.
